std/path.d

```diff
@@ -62,21 +62,39 @@ import std.traits;
 /** String used to separate directory names in a path.  Under
     POSIX this is a slash, under Windows a backslash.
 */
-version(Posix)          enum string dirSeparator = "/";
-else version(Windows)   enum string dirSeparator = "\\";
-else static assert (0, "unsupported platform");
+enum string dirSeparatorPosix = "/";
+enum string dirSeparatorDos = "\\";
 
+/// ditto
+version(Posix)          enum string dirSeparator = dirSeparatorPosix;
+else version(Windows)   enum string dirSeparator = dirSeparatorDos;
+else static assert (0, "unsupported platform");
 
 
 
 /** Path separator string.  A colon under POSIX, a semicolon
     under Windows.
 */
+enum string pathSeparatorPosix = ":";
+enum string pathSeparatorDos = ";";
+
+/// ditto
 version(Posix)          enum string pathSeparator = ":";
 else version(Windows)   enum string pathSeparator = ";";
 else static assert (0, "unsupported platform");
 
+/// Defines how should be built the path, either Posix or Dos style
+enum PathStyle : bool
```

Comment:
We should at least use byte/ubyte to have some room for further systems, even if no one is in the pipeline right now.